CMakeLists.txt

```diff
@@ -1,60 +1,85 @@
-cmake_minimum_required (VERSION 2.8)
-project (NavX)
+cmake_minimum_required(VERSION 2.8.3)
+project(navx)
 set(CMAKE_BUILD_TYPE Release)
-add_definitions(-std=c++11 -Wall -Wextra -Wno-switch)
-
-FIND_PACKAGE (Threads REQUIRED)
 
 file(GLOB SOURCES 
         "${CMAKE_CURRENT_SOURCE_DIR}/src/*.cpp"
         #"${CMAKE_CURRENT_SOURCE_DIR}/include/ahrs/*.h"
 )
 
-include_directories(include)
+## Compile as C++11, supported in ROS Kinetic and newer
+add_definitions(-std=c++11 -Wall -Wextra -Wno-switch)
 
-list(REMOVE_ITEM SOURCES 
-        ${CMAKE_CURRENT_SOURCE_DIR}/src/TimeStamp.cpp
+## Find catkin macros and libraries
+## if COMPONENTS list like find_package(catkin REQUIRED COMPONENTS xyz)
+## is used, also find other catkin packages
+find_package(catkin REQUIRED COMPONENTS
+  sensor_msgs
+  roscpp
 )
+find_package (Threads REQUIRED)
 
-set(HEADER_FILES
-  "include/ahrs/AHRS.h"
-  "include/ahrs/AHRSProtocol.h"
-  "include/ahrs/ContinuousAngleTracker.h"
-  "include/ahrs/IBoardCapabilities.h"
-  "include/ahrs/IIOCompleteNotification.h"
-  "include/ahrs/IIOProvider.h"
-  "include/ahrs/IMUProtocol.h"
-  "include/ahrs/IMURegisters.h"
-  "include/ahrs/InertialDataIntegrator.h"
-  "include/ahrs/ITimestampedDataSubscriber.h"
-  "include/ahrs/OffsetTracker.h"
-  "include/ahrs/SerialIO.h"
-  "include/ahrs/SerialPort.h")
-
-add_library(ahrs SHARED ${SOURCES})
-set_target_properties(ahrs PROPERTIES 
-  PUBLIC_HEADER "${HEADER_FILES}"
-  FRAMEWORK TRUE
+###################################
+## catkin specific configuration ##
+###################################
+## The catkin_package macro generates cmake config files for your package
+## Declare things to be passed to dependent projects
+## INCLUDE_DIRS: uncomment this if your package contains header files
+## LIBRARIES: libraries you create in this project that dependent projects also need
+## CATKIN_DEPENDS: catkin_packages dependent projects also need
+## DEPENDS: system dependencies of this project that dependent projects also need
```

Comment:
If you understand this, you can delete it. Same as the rest of the documentation.